src/java.base/share/classes/java/lang/invoke/VarHandle.java

```diff
@@ -1541,6 +1590,37 @@ VarHandle asDirect() {
     @IntrinsicCandidate
     Object getAndBitwiseXorRelease(Object... args);
 
+    /**
+     * Returns a VarHandle, with access to the same variable(s) as this VarHandle, but whose
+     * invocation behavior of access mode methods is adjusted to
+     * <a href="#invoke-exact-behavior"><em>invoke-exact behavior</em></a>.
+     * <p>
+     * If this VarHandle already has invoke-exact behavior this VarHandle is returned.
+     * @apiNote
+     * Invoke-exact behavior guarantees that that upon invocation of an access mode method
+     * the types and arity of the arguments must match the {@link #accessModeType(AccessMode) access mode type},
+     * otherwise a {@link WrongMethodTypeException} is thrown.
+     *
```

Comment:
While not strictly necessary, the pedantic part to me likes to see assertions like this in the spec:

"Invoking `hasInvokeExactBehavior` on the returned VarHandle is guaranteed to return true."  And conversely, for withInvokeBehavior.